### ISSUE TITLE
Not displaying data on dashboard when it should

### ISSUE THREAD

**Issue by aleglez** [metabase/metabase#9893]
Hi, I wanted to report a bug I found.

I’m having an issue trying to display results for a query that is actually not returning anything.
The problem is I have 3 queries on the same graph in a dashboard, and the one returning empty is the 1st one so Metabase cant handle that properly and is not showing the results from the other 2 either. This happens when a filter is used and the 1st query have no results for the selected value.

- Browser: Chrome 73.0.3683.103
- Operating system: Windows 10
- Database: Postgres
- Metabase version: 0.32.4
- Metabase hosting environment: Heroku
- Metabase internal database: Postgres

I was able to reproduce it using your Sample Database:

Create the next 2 queries:

- Test 2018 orders (H2 Sample Dataset)

```
SELECT SUM(ORDERS.QUANTITY) as Amount, to_char(ORDERS.CREATED_AT, 'Mon') as Mon
FROM ORDERS JOIN PEOPLE ON ORDERS.USER_ID = PEOPLE.ID
WHERE YEAR(ORDERS.CREATED_AT) = YEAR(current_date - 365) [[AND {{User}}]]
GROUP BY to_char(ORDERS.CREATED_AT, 'Mon'), extract(year from ORDERS.CREATED_AT), extract(month from ORDERS.CREATED_AT)
ORDER BY extract(year from ORDERS.CREATED_AT), extract(month from ORDERS.CREATED_AT)
```

- Test 2019 orders (H2 Sample Dataset)

```
SELECT SUM(ORDERS.QUANTITY) as Amount, to_char(ORDERS.CREATED_AT, 'Mon') as Mon
FROM ORDERS JOIN PEOPLE ON ORDERS.USER_ID = PEOPLE.ID
WHERE YEAR(ORDERS.CREATED_AT) = YEAR(current_date) [[AND {{User}}]]
GROUP BY to_char(ORDERS.CREATED_AT, 'Mon'), extract(year from ORDERS.CREATED_AT), extract(month from ORDERS.CREATED_AT)
ORDER BY extract(year from ORDERS.CREATED_AT), extract(month from ORDERS.CREATED_AT)
```

Use bar graph on both and add them to a dashboard graph (2018 first), also using bars, it will look like this:

![screenshotAtUploadCC_1555686985815](https://user-images.githubusercontent.com/20310019/57246322-00177800-700b-11e9-9c76-d457a0ef9830.png)

Then add a filter for the users on the dashboard and use "Abelardo Purdy". The thing is that user has orders for 2019 but not for 2018, so the expected behavior should be to only display results for 2019, but this is what happens:

![screenshotAtUploadCC_1555686985816](https://user-images.githubusercontent.com/20310019/57246334-06a5ef80-700b-11e9-90b6-770fea7b866f.png)

This is exactly the same problem I'm having.

**Comment by flamber** [comment]
Just for reference, this was reproduced in [the forum](https://discourse.metabase.com/t/need-a-way-to-display-empty-results/5988).

**Comment by paulrosenzweig** [comment]
Resolved by #10178
